channels/ON-CHAIN.md

```diff
@@ -363,15 +374,144 @@ channel.
+in a different channel `state_hash`.
+
+The payload can be either empty or a signed transaction.
+If the payload is empty - the last on-chain persisted state is used. In this case the proof of inclusion root has of the proof of inclusion must be equal to the one persisted for the channel on-chain. The round being used is the one stored in the channel on-chain.
+If the payload is a transaction it MUST be a channel_offchain_tx. It MUST be co-signed.
+
+An off-chain transaction payload is a valid transaction if it has:
+* `state_hash` equal to the proof of inclusion's root hash. This is a proof
+  that the PoI is a correct one
+* `channel_id` being the same as the transaction `channel_id`
+* `round` being greater to the last on-chain `round` for that channel id.
+
+The solo_payload can not be empty. It MUST be a signed transaction.
+It MUST be a channel_offchain_tx
+
+An off-chain transaction solo payload is a valid transaction if it has:
+* `channel_id` being the same as the transaction `channel_id`
+* `round` being the incremented by one `round` provided in the `payload`. 
+* `updates` has a single update that is a call to a contract. The caller of
+  this update MUST be the poster of the force progress transaction. `gas` and
+  `gas_price` are specified. The gas fees are going to be paid by the poster
+  of the transaction.
+* `state_hash` will be the root hash of the updated channel's state trees.
+  After applying the contract call to the provided `poi` and updating accounts
+  accodingly - a new channel's state tree is produced. It SHOULD have the same
+  root hash as the one provided in the `solo_payload`
+
+Serialization defined [here](../serializations.md#channel-solo-force-progress-transaction)
+
+### Force progress side effects
+
+#### Updating channel object
+
+A channel state trees are recreated according to the `poi` being provided. The
+update provided in the `updates` of the `solo_payload` is a contract call one.
+It is applied on the channel's state trees and modifies them. The modified
+has a root hash. It might be:
+
+- equal to the `state_hash` provided in the `solo_payload`. This hash
+  indeed is the expected result of the contract call and the blockchain had
+  confirmed it. The on-chain channel object is updated accodingly:
+  - channel's state hash is updated to be the newly computed one
+  - channel's round is the one in the `solo_payload`
+  - if the channel had been in a closing state - closing balances of
+    participants are updated according to the ones in the modified channel state trees
+
+- no equal to the `state_hash` provided in the `solo_payload`. The hash
+  provided was not confirmed to be the expected one. The on-chain channel
+  object is NOT modified
+
+#### Call object
+
+If the `channel_force_progress_tx` is a valid one - the contract call in the
+`solo_payload`'s `updates` is executed. This consumes gas. The `update` itself
+defines both the gas limit and the gas price. After the contract call is being
```

Comment:
After the contract call **has been** executed and the real gas consumption **has** been calculated, [comma] the balance ...